HW4_Toropov/protein_tools.py

```diff
@@ -0,0 +1,336 @@
+
+
+def is_rna(seq: str):
+    """
+    Check the existence of a RNA sequence, return boolean.
+    """
+    unique_chars = set(seq.upper())
+    return unique_chars <= alphabet_rna
+
+
+def compute_molecular_weight(protein: str) -> tuple:
+    """
+    Compute molecular weight (g/mol) of protein sequence.
+
+    Argument:
+    - protein (str): protein sequence.
+
+    Return:
+    - tuple with protein sequence and computed molecular
+    weight (float rounded to 3 decimal places).
+    """
+    molecular_weight = 0
+    for amino_acid in protein.upper():
+        molecular_weight += amino_acid_masses[amino_acid]
+    return protein, round(molecular_weight, 3)
+
+
+def compute_length(protein: str) -> tuple:
+    """
+    Compute the length of the input protein sequence.
+
+     Argument:
+    - protein (str): protein sequence.
+
+    Return:
+    - tuple with protein sequence and computed length.
+    """
+    return protein, len(protein)
+
+
+def protein_to_dna(protein: str) -> str:
+    """
+    Returns possible variants of DNAs for a given protein sequence.
+
+    Argument:
+    - protein (str): protein sequence.
+
+    Return:
+    - string, variants of nucleic acids.
+    If several codons correspond to a given amino acid they are displayed with a '/'.
+
+    Does not distinguish between lowercase and uppercase letters.
+
+    Examples:
+
+    -'MACDRS' -> 'ATG GCT/GCC/GCA/GCG TGT/TGC GAT/GAC CGT/CGC/CGA/CGG/AGA/AGG TCT/TCC/TCA/TCG/AGT/AGC'
+    -'MaCdrS' -> 'ATG GCT/GCC/GCA/GCG TGT/TGC GAT/GAC CGT/CGC/CGA/CGG/AGA/AGG TCT/TCC/TCA/TCG/AGT/AGC'
+    """
+    nucleic_acid_seq = ""
+
+    for aa in protein.upper():
+        codons = dna_codons.get(aa)
+        nucleic_acid_seq += "/".join(codons) + " "
+
+    return nucleic_acid_seq[:-1]
+
+
+def count_amino_acids(protein: str) -> dict:
+    """
+    Calculates the number of each aminoacid in a given protein sequence.
+
+    Argument:
+    - protein (str): protein sequence.
+
+    Return:
+    - dictionary, where a key is the aminoacid letter and value is number of this aminoacid.
+
+    Does not distinguish between lowercase and uppercase letters.
+
+    Examples:
+
+    -'MACDRS' -> {'M': 1, 'A': 1, 'C': 1, 'D': 1, 'R': 1, 'S': 1}
+    -'MaCdrS' -> {'M': 1, 'A': 1, 'C': 1, 'D': 1, 'R': 1, 'S': 1}
+    """
+    amino_acids_dict = {}
+    for aa in protein.upper():
+        if aa in amino_acids_dict:
+            amino_acids_dict[aa] += 1
+        else:
+            amino_acids_dict[aa] = 1
+    return amino_acids_dict
+
+
+def compute_hydrophobicity(protein: str) -> tuple:
+    """
+    Compute the percentage of gydrophobic aminoacids in protein sequence.
+
+    Argument:
+    - protein (str): protein sequence. Includes hydrophobic
+    and hydrophilic aminoacids.
+
+    Return:
+    - tuple with protein sequence and computed percentage
+    of gydrophobic aminoacids.
+    """
+    count_of_gydrophobic = 0
+    for i in range(len(protein)):
+        if protein[i] in gydrophobic_aminoacids:
```

Comment:
```suggestion
    for aa in protein:
        if aa in gydrophobic_aminoacids:
```

Можно итерироваться по индексам там, где нужны именно элементы